docs/common-questions.md

```diff
@@ -11,25 +11,34 @@ If you need `async fetchOptions` you can add an exchange that looks like this:
 
 ```js
 import { makeOperation } from '@urql/core';
+import { Exchange, Operation } from 'urql';
+import { pipe, mergeMap, map, fromPromise } from 'wonka';
 
-export const fetchOptionsExchange = (fn: any): Exchange => ({ forward }) => ops$ => {
-  return pipe(
-    ops$,
-    mergeMap((operation: Operation) => {
-      const result = fn(operation.context.fetchOptions);
-      return pipe(
-        typeof result.then === 'function' ? fromPromise(result) : fromValue(result),
-        map((fetchOptions: RequestInit | (() => RequestInit)) => {
-          return makeOperation(operation.kind, operation, {
-            ...operation.context,
-            fetchOptions,
-          });
-        })
-      );
-    }),
-    forward
-  );
-};
+export const fetchOptionsExchange =
+  (fn: (fetchOptions: RequestInit) => Promise<RequestInit>): Exchange =>
```

Comment:
to avoid issues we initially made this `(fetchOptions: RequestInit) => Promise<RequestInit> | RequestInit` and had a ternary that did this in the stream:

```
const result = fn(operation.context.fetchOptions);
return pipe(
  typeof result.then === 'function' ? fromPromise(result) : fromValue(result),
);
```

This mainly to avoid introducing a footgun 😅

Reply:
Yep, sounds good. Fixed! Thanks for the tip 👍